### DISCUSSION TITLE
Backtransformation of Estimates

### DISCUSSION THREAD

**Discussion by lvclarke** [inlabru-org/inlabru#204] [General]
Hi,

I wonder if you're able to provide some help on how to correctly backtransform LGCP model outputs?

Specifically, if you wanted to outline how density varies in relation to covariates within the model, what would be the correct transformation to describe the fixed effect estimates - ie. As distance increases by 1km, density increase by X individuals/km, or similar?

For example, the tutorial below is helpful in plotting how elevation affects log(density), but would be really helpful to understand how you could backtransform the estimates to produce a snappy summary of the relationship? 
https://inlabru-org.github.io/inlabru/articles/2d_lgcp_covars.html

Thanks!

**Comment by finnlindgren** [comment]
The most important aspect is that effects that are linear and act in an additive way in log-scale (i.e. the linear predictor) are  "log-linear" and multiplicative on the "natural" scale.  I.e. if the elevation coefficient is $\beta$, it means that "density at elevation (A+\delta)" =  \exp(\beta * \delta) * "density at elevation A"; a unit change in elevation gives a multiplicative change in density by the unitless number $\exp(\beta)$, when keeping all else fixed.   On the other hand, _derivative_ of density w.r.t. elevation at a specific altitude is $\lim_{\delta\rightarrow 0}(\exp(\beta(A+\delta)+\text{TheRest}) - \exp(\beta A+\text{TheRest}))/\delta=\beta\exp(\beta A+\text{TheRest})$, which is in units density per [unit elevation], but depends on the reference elevation $A$ and TheRest, i.e. all the other contributions to the model.

So there's no "free lunch" here; either one deals with the raw coefficients, the multiplicative change, or the derivative at a reference level when keeping all else fixed, where the third option gives a different value for each reference level and values of the rest of the model.  The "multiplicative change" interpretation is probably the clearest, as it describes the ratio between two density values at a unit difference in the covariate, keeping all else fixed, regardless of what the rest of the model is fixed to.

**Reply by finnlindgren** [reply]
One can of course also "invert" the multiplicative interpretation, e.g with $2=\exp(\beta \delta_2)$ giving $\delta_2=\ln(2) / \beta$, where $\delta_2$ can be interpreted as the covariate change needed to give a doubling of the density (for a negative value, $|\delta_2|$ would be interpreted as the change needed to _halve_ the density).

Note that these are all non-linear relationships, so be careful when computing their posterior distribution properties; quantiles transform nicely in monotonic invertible transformations, but expectations and variances do not.  So if you have quantiles for $\beta$, you can use them to get quantiles for e.g. $\delta_2$, but for posterior moments you'll need to use a `predict()` call.
